packages/shared/lib/services/sync/data/records.service.ts

```diff
@@ -549,3 +545,35 @@ export async function getRecordsByExternalIds(external_ids: string[], nango_conn
 
     return result as unknown as SyncDataRecord[];
 }
+
+export async function findSyncsWithDeletableRecords(limit: number): Promise<Pick<Sync, 'id'>[]> {
+    return db.knex
+        .select<Pick<Sync, 'id'>[]>('syncs.id')
+        .from('_nango_sync_data_records AS rec')
+        .join('_nango_syncs AS syncs', 'syncs.id', '=', 'rec.sync_id')
+        .where('syncs.deleted', true)
+        .groupBy('syncs.id')
+        .limit(limit);
+}
+
+export async function deleteRecordsBySyncIdSafe(syncId: string, limit: number = 5000): Promise<void> {
```

Comment:
I'm not sure what `Safe` means in this context. Can a better name be used here?

Reply:
it was just to emphasis that it was okay to use vs the regular delete, but I can remove the Safe entirely since there is no alternative

Reply:
Done ✅